Crash in search function

After upgrading from `0.3.9` to `0.3.9-0005` and running for several hours, Yggdrasil crashed.
I have restarted it and will try to catch exception.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x5b99fa]

goroutine 948681 [running]:
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*Conn).search.func1.1(0
x0, 0x6bef60, 0xc00031b910)
        /home/circleci/project/src/yggdrasil/conn.go:121 +0x9a
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*searchInfo).doSearchSt
ep(0xc00004a4e0)
        /home/circleci/project/src/yggdrasil/search.go:134 +0x2de
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*searchInfo).continueSe
arch(0xc00004a4e0)
        /home/circleci/project/src/yggdrasil/search.go:150 +0x32
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*searchInfo).continueSe
arch.func1.1()
        /home/circleci/project/src/yggdrasil/search.go:161 +0x74
github.com/Arceliar/phony.(*Inbox).run(0xc000097a28)
        /go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20190907031509-af5bdbeecab
6/actor.go:96 +0x94
created by github.com/Arceliar/phony.(*Inbox).restart
        /go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20190907031509-af5bdbeecab
6/actor.go:127 +0x46
```

I have a crash on startup, every time.

2019/10/05 23:39:59 Build name: yggdrasil-develop
2019/10/05 23:39:59 Build version: 0.3.9-0019
2019/10/05 23:39:59 Starting up...
2019/10/05 23:39:59 Starting switch
2019/10/05 23:39:59 Starting router
2019/10/05 23:39:59 Startup complete
... <1s elapses
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x5b1c63]

goroutine 5133 [running]:
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*Conn).search.func1.1(0x0, 0x6ca5e0, 0xc000079870)
        /home/circleci/project/src/yggdrasil/conn.go:121 +0x93
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*searchInfo).checkDHTRes(0xc000080b60, 0xc0000b03f0, 0x1f6135a8ddec6
f01)
        /home/circleci/project/src/yggdrasil/search.go:208 +0x294
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*searchInfo).handleDHTRes(0xc000080b60, 0xc0000b03f0)
        /home/circleci/project/src/yggdrasil/search.go:81 +0x78
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*dht).handleRes(0xc0000c3a50, 0xc0000b03f0)
        /home/circleci/project/src/yggdrasil/dht.go:250 +0x13d
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*router)._handleDHTRes(0xc0000c3a18, 0xc0000eef00, 0x47, 0x2f1, 0xc0
00749ed0)
        /home/circleci/project/src/yggdrasil/router.go:242 +0x9a
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*router)._handleProto(0xc0000c3a18, 0xc0002fbc00, 0xb5, 0x301)
        /home/circleci/project/src/yggdrasil/router.go:204 +0x372
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*router)._handlePacket(0xc0000c3a18, 0xc0002fbc00, 0xb5, 0x301)
        /home/circleci/project/src/yggdrasil/router.go:145 +0x91
github.com/yggdrasil-network/yggdrasil-go/src/yggdrasil.(*router).handlePackets.func1()
        /home/circleci/project/src/yggdrasil/router.go:103 +0x7e
github.com/Arceliar/phony.(*Inbox).run(0xc0000c3a18)
        /go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191005181740-21679e75e3f0/actor.go:104 +0x8d
created by github.com/Arceliar/phony.(*Inbox).restart
        /go/pkg/mod/github.com/!arceliar/phony@v0.0.0-20191005181740-21679e75e3f0/actor.go:130 +0x3f

--- a/src/yggdrasil/conn.go
+++ b/src/yggdrasil/conn.go
@@ -115,12 +115,12 @@ func (c *Conn) search() error {
 				default:
 					if sinfo != nil {
 						// Finish initializing the session
-						sinfo.setConn(nil, c)
-					}
-					c.session = sinfo
-					c.nodeID = crypto.GetNodeID(&c.session.theirPermPub)
-					for i := range c.nodeMask {
-						c.nodeMask[i] = 0xFF
+						c.session = sinfo
+						c.session.setConn(nil, c)
+						c.nodeID = crypto.GetNodeID(&c.session.theirPermPub)
+						for i := range c.nodeMask {
+							c.nodeMask[i] = 0xFF
+						}
 					}
 					err = e
 					close(done)

